### DISCUSSION TITLE
IMediaPathScheme should receive the IContentBase, not just the GUIDs.

### DISCUSSION THREAD

**Discussion by LennardF1989** [umbraco/Umbraco-CMS#14337] [Features and ideas]
`IMediaPathScheme` now gets the content.Key and propertyType.key from the MediaFileManager, but why not just pass the whole `content`? This would prevent all kinds of additional lookups and also allows the manipulation of the content before it's eventually persisted.

This has remained unchanged since v7 IIRC.

**Comment by LennardF1989** [comment]
The associated use-case here: we want to be able to decide where to store a media item based on the parent media item it belongs to. That way, for example, we could store and retrieve media items in/from different blob storage containers.

Currently, `IMediaPathScheme` will get a propertyType.key for a media item that may not be persisted yet, so it's impossible to retrieve the parent. Passing the IMedia item itself would solve the issue (for this use case).

**Comment by bergmania** [comment]
Should be okay to make these changes, as long as we do not introduce breaking changes..

I guess that should be possible if the Umbraco implementation just calls the old signature.

**Reply by LennardF1989** [reply]
Please see https://github.com/umbraco/Umbraco-CMS/pull/14338 and let me know what you think :) I just carefully read Sebastiaans note on backwards compatibility, but curious what you think of the current implementation (setting backwards compatibility aside for a moment). Do you see any other opportunities while we are at it? Something I missed?

**Reply by bergmania** [reply]
I added a few comments to the PR.. Overall I think it looks good, but i'm concerned about the nullability is some cases..
